### PR TITLE
Increase the logstash limit to 100 in web and 500 in CLI contexts and set 'http_user_agent' based on _SERVER['HTTP_USER_AGENT']

### PR DESCRIPTION
**Pull request by rinatkhaziev**
## Description

Currently, we have a cut-off of 30 logstash entries per request, as a failsafe. The problem is we call trigger_error when we're over the threshold. This leads to an unhelpful situation where a useful log entry gets replaced by an unuseful generic error not providing any information beyond the fact that it happened.

This PR increases the limit to 100 for web requests and 500 for CLI requests - this may change pending feedback.
When a threshold is crossed only one E_USER_WARNING is triggered.

Additionally, this PR also surfaces user agent for request (if available) 

### Library updated: Logger

We've updated the internal logging library to surface the user agent information for custom log entries and to increase the maximum number of logs to be processed during a single request.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [x] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [x] I've created a changelog description that aligns with the provided examples.

### CODE REVIEW

File: logstash/class-logger.php [hunk 1]
```diff
@@ -535,4 +540,28 @@ public static function wp_debug_log( array $entry ) : void {
...
+	 */
+	public static function get_user_agent() {
+		if ( defined( 'WP_CLI' ) && WP_CLI ) {
+			return 'cli';
+		} else {
+			// Match handling in other places
+			// phpcs:ignore WordPressVIPMinimum.Variables.RestrictedVariables.cache_constraints___SERVER__HTTP_USER_AGENT__
+			return substr( sanitize_text_field( $_SERVER['HTTP_USER_AGENT'] ?? '' ), 0, 255 );
+		}
+	}
```
Comment (WPprodigy) [inline logstash/class-logger.php:566]: Small nitpick, but removes the `else` :)

```suggestion
	public static function get_user_agent() {
		if ( defined( 'WP_CLI' ) && WP_CLI ) {
			return 'cli';
		}
		
		// Match handling in other places
		// phpcs:ignore WordPressVIPMinimum.Variables.RestrictedVariables.cache_constraints___SERVER__HTTP_USER_AGENT__
		return substr( sanitize_text_field( $_SERVER['HTTP_USER_AGENT'] ?? '' ), 0, 255 );
	}
```

**Review by WPprodigy — COMMENTED** [review]
So the 101st entry will fail and log the warning. But the 100 prev ones will still be logged right?

**Comment by rinatkhaziev** [comment]
> So the 101st entry will fail and log the warning. But the 100 prev ones will still be logged right?

Yep, that logic doesn't change, what's changing is the limit and not logging the useless message over and over, instead only log once.

File: logstash/class-logger.php [hunk 2]
```diff
@@ -290,19 +291,17 @@ public function debug( string $feature, string $message, array $context = [] ) :
...
+			'blog_id'         => get_current_blog_id(),                       // Required.
+			'http_host'       => strtolower( $_SERVER['HTTP_HOST'] ?? '' ),   // phpcs:ignore -- Optional.
+			'severity'        => '',                                          // Optional.
+			'feature'         => '',                                          // Required.
+			'message'         => '',                                          // Required.
+			'user_id'         => get_current_user_id(),                       // Optional.
+			'extra'           => [],                                          // Optional.
+			'timestamp'       => gmdate( 'Y-m-d H:i:s' ),                     // Required.
+			'index'           => 'log2logstash',                              // Required,
+			'http_user_agent' => self::get_user_agent(),
```
Comment (WPprodigy) [inline logstash/class-logger.php:304]: Just to keep the pattern of the rest (assuming it is optional):

```suggestion
			'timestamp'       => gmdate( 'Y-m-d H:i:s' ),                     // Required.
			'index'           => 'log2logstash',                              // Required.
			'http_user_agent' => self::get_user_agent(),                      // Optional.
```

**Review by WPprodigy — APPROVED** [review]
Small commenting change, otherwise looks good

**Review by WPprodigy — APPROVED** [review]

**Comment by rinatkhaziev** [comment]
r2063-stacks

**Pull request merged**